### ISSUE TITLE
Suggestion: add rule for `!isTRUE` -> `isFALSE`

### ISSUE THREAD

**Issue by m-muecke** [r-lib/lintr#2607]

**Comment by MichaelChirico** [comment]
Good idea. I think it can be part of `comparison_negation_linter()`.

**Comment by m-muecke** [comment]
Ah, I guess I forget of the use-case where `NULL` or `NA` wants be checked:

``` r
!isTRUE(NULL) == isFALSE(NULL)
#> [1] FALSE
```

``` r
!isTRUE(NA) == isFALSE(NA)
#> [1] FALSE
```

<sup>Created on 2024-06-14 with [reprex v2.1.0](https://reprex.tidyverse.org)</sup>

**Comment by MichaelChirico** [comment]
Oh, true, more generally,

```r
isTRUE(x) <=> length(x) == 1 && is.logical(x) && !is.na(x) && x
isFALSE(x) <=> length(x) == 1 && is.logical(x) && !is.na(x) && !x
```

So `!isTRUE(x) != isFALSE(x)` whenever `length(x) !=1 || !is.logical(x) || is.na(x)`.

So actually I'm not sure there's anything we can do here statically.
